### PR TITLE
[TT-Train] Added Yaml Configs support

### PR DESCRIPTION
**Pull request by dmakoviichuk-tt**
### What's changed
* Moved models out of examples.
* Added yaml configs support
* A few small improvements

### Checklist
- [x] Post commit CI passes
- [x] Blackhole Post commit (if applicable)
- [x] Model regression CI testing passes (if applicable)
- [x] Device performance regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tt-train/sources/ttml/ops/dropout_op.cpp [hunk 1]
```diff
@@ -16,6 +16,9 @@
 namespace ttml::ops {
 
 autograd::TensorPtr dropout(const autograd::TensorPtr& tensor, float probability) {
+    if (probability == 0.0F) {
```
Comment (dmakoviichuk-tt) [inline tt-train/sources/ttml/ops/dropout_op.cpp:19]: @rfurko-tt

File: tt-train/tests/3rd_party/tokenizers_test.cpp [hunk 2]
```diff
@@ -15,8 +15,8 @@ using tokenizers::Tokenizer;
 namespace {
 
 std::string getTestDataDir() {
-    const char* envVar = std::getenv("TEST_DATA_DIR");
-    return (envVar) ? std::string(envVar) : std::string(TEST_DATA_DIR);
+    const char* env_var = std::getenv("TEST_DATA_DIR");
```
Comment (dmakoviichuk-tt) [inline tt-train/tests/3rd_party/tokenizers_test.cpp:18]: @rfurko-tt

File: tt-train/sources/examples/mnist_mlp/main.cpp [hunk 3]
```diff
@@ -55,27 +56,46 @@ float evaluate(DataLoader &test_dataloader, Model &model, size_t num_targets) {
...
+    config.weight_decay = training_config["weight_decay"].as<float>();
+    config.is_eval = training_config["is_eval"].as<bool>();
+    config.model_save_interval = training_config["model_save_interval"].as<int>();
+    config.mlp_config = ttml::models::mlp::read_config(training_config["mlp_config"]);
+    return config;
+}
 
-    app.add_option("-n,--num_epochs", num_epochs, "Number of epochs")->default_val(num_epochs);
-    app.add_option("-s,--model_path", model_path, "Model path")->default_val(model_path);
-    app.add_option("-e,--eval", is_eval, "eval only mode")->default_val(is_eval);
```
Comment (dmakoviichuk-tt) [inline tt-train/sources/examples/mnist_mlp/main.cpp:76]: return back eval to cmd.

**Review by rfurko-tt — APPROVED** [review]
LGTM

**Pull request merged**